target/bin/listmailuser

```diff
@@ -16,19 +16,35 @@ function dovecot_quota_to_hr()
 DATABASE="/tmp/docker-mailserver/postfix-accounts.cf"
 ALIASES="/tmp/docker-mailserver/postfix-virtual.cf"
 
+if [[ -z ${ENABLE_QUOTAS} ]]
+then
+  # container was started without the env ENABLE_QUOTAS set.
+  # get default value from .bashrc
+  if grep -qE "^export ENABLE_QUOTAS='0'$" /root/.bashrc
+  then
+    ENABLE_QUOTAS=0
+  else
+    ENABLE_QUOTAS=1
+  fi
+fi
+
```

Comment:
> or better yet via an entrypoint script that will ensure we apply ENV defaults upfront?

😄 As of now, there is no entrypoint script. I wanted to quickly fix an error and not introducing a complete new bigger feature.

> Any subshell or subprocess invoked afterwards should inherit that, unless invoked via a non-interactive shell (-c)?

No. It's the other way around. Only when starting bash as an interactive shell, `.bashrc` is automatically sourced.

We could just source that file manually in this case for example. But in general, `.bashrc` can contain other things like functions, aliases and normal commands. That's why I tried to avoid sourcing it. In my setup for example, there are some handy aliases and other things in that file.

> For now, maintain this as a separate file (eg: /target/scripts/init-default-env.sh) and source that

I tried to avoid introducing another file. But if everyone is fine with that, I can implement it. That would probably the cleanest/best solution 👍 

> Any defaults/adjustments that the mailserver service makes and stores to .bashrc won't be visible to all other services?

Correct.

> An entrypoint script will run when starting a container with a command such as setup, which would not run all of our initialization startup scripts (including the ones that create .bashrc here). Presently our entrypoint is just dumb-init.

Using an entrypoint script is not a good idea, especially when running multiple services in one container as we do. An (dumb) init system [takes care of a few things](https://github.com/Yelp/dumb-init#why-you-need-an-init-system).

BTW: I wanted to link to the dumb-init PR where I thought there were already discussions about pro/cons. I found the [commit](https://github.com/docker-mailserver/docker-mailserver/commit/6e9f6152becf10938c41934014fd877768d871c2), but not the PR?

Reply:
> I wanted to quickly fix an error and not introducing a complete new bigger feature.

I understand, that's why I suggested extracting it to a centralized file so this sort of thing doesn't pop up elsewhere in future.

> No. It's the other way around. Only when starting bash as an interactive shell, `.bashrc` is automatically sourced.

That's what I meant :sweat_smile: 

`-c` is non-interactive shell, and doesn't source `.bashrc`.

> I tried to avoid introducing another file. But if everyone is fine with that, I can implement it.

I think that would be best, feel free to see how others weigh in :)

> Using an entrypoint script is not a good idea, especially when running multiple services in one container as we do.

Isn't it effectively what `start-mailserver.sh` / `mailserver` (supervisor service) is?

We call the `dumb-init` that takes the `supervisord` CMD and that begins the `mailserver` service. Only difference is that if you use `docker run ... setup ...` instead of `docker exec` on a running container, the CMD will override `supervisord` instead of doing any preprocessing via ENTRYPOINT (_which can optionally start the mailserver service along with dumb-init still_).

> An (dumb) init system takes care of a few things.

For Docker at least, [an alternative `tini` is used by default](https://docs.docker.com/engine/reference/run/#specify-an-init-process) if using `--init`/`init:` ([docker docs init page](https://docs.docker.com/config/containers/multi-service_container/)), but for `docker-mailserver` I guess it makes more sense to bake it into the Dockerfile as part of the entrypoint :sweat_smile: .

> BTW: I wanted to link to the dumb-init PR where I thought there were already discussions about pro/cons. I found the commit, but not the PR?

That looks like roughly the time the project transferred to an org, where PR history seemed to be a bit iffy and @georglauterbach just carried over some changes manually I think. I don't see the PR looking back around Jan 2021, looks like it might have just been lost and the change just directly committed to master (_I don't think we had any branch protection back then), issues and PR references weren't accurate as a result.

Reply:
> Isn't it effectively what start-mailserver.sh / mailserver (supervisor service) is?

No. Supervisor is not an init system. See my link above for the differences and what an init process additionally does compared to any script.

> For Docker at least, an alternative tini is used by default if using --init/init

There isn't much difference between those two solutions in what they do. However, `--init` depends on docker. While our solution is build-in and works everywhere. Only "downside" is a few more used bytes in our image 😄

As you and Georg are fine with a new file, I am going to implement it 🎉

Reply:
> No. Supervisor is not an init system. See my link above for the differences and what an init process additionally does compared to any script.

Misunderstanding. I was referring to how this differs:

a. `ENTRYPOINT=dumb-init` + `CMD=supervisord` (or any override for CMD)
b. `ENTRYPOINT=entrypoint.sh` (that calls `dumb-init` with CMD) + `CMD=supervisord` (or any override for CMD)

Both still use `dumb-init`, one just allows running some conditional logic with the initial shell that runs the entrypoint (_that currently is equivalent to an `entrypoint.sh` that just calls `dumb-init` with `$@` to pass the CMD to_).

Reply:
Entrypoint = PID1. If entrypoint is a bash script, it will not take care of signal forwarding, reaping processes etc. An init-system must be PID1.

But I think we can skip this discussion. If a script like `listmailuser` needs ENVs, it can just source `/etc/dms-settings` now 😎